packages/ifpack2/src/Ifpack2_Relaxation_def.hpp

```diff
@@ -143,10 +143,22 @@ namespace {
       return Teuchos::ScalarTraits<Scalar>::eps ();
     }
   };
+
+
+
 } // namespace (anonymous)
 
 namespace Ifpack2 {
 
+template<class MatrixType>
+void Relaxation<MatrixType>::updateCachedMultiVector(const Teuchos::RCP<const Tpetra::Map<local_ordinal_type,global_ordinal_type,node_type> > & map, size_t numVecs) const{
+  // Allocate a multivector if the cached one isn't perfect
+  // Note: We check for map pointer equality here since it is much cheaper than isSameAs()
+  if(cachedMV_.is_null() || &*map != &*cachedMV_->getMap() || cachedMV_->getNumVectors() !=numVecs) 
```

Comment:
Map equality operator means isSameAs, right?  I'd rather reallocate a MV than potentially do a collective.  At least maybe put that test at the end so it's easy to fall through without forcing a collective.

Reply:
Pointer equality is the first think isSameAs  checks.